internal/jet/range_expression.go

```diff
@@ -0,0 +1,90 @@
+package jet
+
+// RangeExpression is interface for date range types
+type RangeExpression interface {
```

Comment:
To be honest, the reason I didn't use generics in that part of the code was because your coding style didn't include any generic functions. However, after receiving your comment, I will definitely add generics to that part because it's the best place to use them

Reply:
> And the main change will be golang version, which right now its 1.11 and has to be updated to 1.18 at least. Are you sure about it?
Yeah, that's fine.

Reply:
Could you take a look at current state, please? and If it is possible provide more detail about this part, I like to know about your idea

Reply:
Current state looks good. Now you just need to add a template to `RangeExpression`. Something like:

```golang
type Range[T Expression] interface {
	Expression

	EQ(rhs Range[T]) BoolExpression
        ....
	CONTAINS(rhs T) BoolExpression
	OVERLAP(rhs Range[T]) BoolExpression
        ...
}
```
After this, you'll probably have to add the same template to each of the new range functions and structs.
And eventually in `postgres` package you can export new types:

```golang
// new expressions
type Int4RangeExpression = jet.Range[IntegerExpression]
type Int8RangeExpression = jet.Range[IntegerExpression]
// ....
// new columns
var Int4RangeColumn = jet.Range[IntegerExpression]
type ColumnInt4Range = jet.ColumnRange[IntegerExpression]
// ...
// etc...
```